driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -2043,8 +2117,16 @@ public void run() {
                         schemaInAgreement = ControlConnection.waitForSchemaAgreement(connection, Manager.this);
                         if (!schemaInAgreement)
                             logger.warn("No schema agreement from live replicas after {} s. The schema may not be up to date on some nodes.", configuration.getProtocolOptions().getMaxSchemaAgreementWaitSeconds());
-                        if (refreshSchema)
-                            ControlConnection.refreshSchema(connection, keyspace, table, Manager.this, false);
+                        if (refreshSchema) {
+                            submitSchemaRefresh(keyspace, table);
+                            // force synchronous schema refresh
+                            Uninterruptibles.getUninterruptibly(schemaRefreshRequestDebouncer.deliverEvents());
```

Comment:
This mimics the old, synchronous behavior where ControlConnection.refreshSchema was called directly.